docassemble/ALWeaver/custom_values.py

```diff
@@ -1,105 +1,160 @@
 from typing import List, Union
 from pathlib import Path
-import yaml
-from docassemble.base.util import log, DADict
+import ruamel.yaml as yaml
+from docassemble.base.util import log, DADict, DAList, user_info, DAStore, path_and_mimetype
 from docassemble.base.core import objects_from_file
 # TODO(brycew): is this too deep into DA? Unclear if there are options to write
 # sources files to a package while it's running.
+import ruamel.yaml as yaml
+import sys
 from docassemble.base.functions import package_data_filename
+from packaging.version import Version
+import os
+from more_itertools import unique_everseen
 
-"""
-Container for widely used data that may be altered by user inputs.
-"""
+################# To refactor - I don't think these are used but they are mentioned in interview_generator.py
+class Object(object):
+  pass
+custom_values = Object()
+custom_values.people_plurals_map = {}
+custom_values.org_specific_config = None
+########################## End to refactor
 
 __all__ = ['get_possible_deps_as_choices', 'get_pypi_deps_from_choices', 
-           'get_yml_deps_from_choices']
+           'get_yml_deps_from_choices', 'SettingsList', 'load_capabilities',
+          'advertise_capabilities', '_package_name']
 
-# This is to workaround fact you can't do local import in Docassemble playground
-class Object(object):
-  pass
+class SettingsList(DAList):
+  """
+  A simple list that can sync itself to a DAStore
+  """
+  def init(self, *pargs, **kwargs):
+    super().init(*pargs, **kwargs)
 
-custom_values = Object()
+  def hook_after_gather(self):
+    if hasattr(self, 'store'):
+      self.store.set(self.instanceName, self)
+  
+  def __str__(self):
+    return "\n".join(self.complete_elements())
 
-# Var names of people-type lists that developers add while using the weaver.
-# Could be a list, but a dict will match RESERVED_PLURALIZERS_MAP in the
-# places it's used, so maybe easier to treat them both the same.
-custom_values.people_plurals_map = {}
+def load_capabilities(base:str="docassemble.ALWeaver", minimum_version="1.5", include_playground=False):
+  """
+  Get the contents of the current package's capabilities file.
```

Comment:
A little confused here: this function will load the capabilities of all the packages who have "advertised", right? Not just the current package? Are capabilities just configuration options / choices? Looking at everything together capabilities sounds like it's the Weaver telling other packages what attributes they can have / need to have in their `configuration_capabilities.yml`, which isn't what it's doing.

Reply:
Good point, I'll clarify this docstring.